src/main/java/ir/sahab/dockercomposer/DockerComposeRunner.java

```diff
@@ -250,4 +247,40 @@ private String getContainerIp(String containerId) {
                 "{{range .NetworkSettings.Networks}}{{.IPAddress}}{{end}}", containerId);
         return result.outputString().trim();
     }
+
+    /**
+     * Returns container info by using "{@code docker ps}" command. The output format is
+     * CONTAINER_NAME#CONTAINER_PORTS.
+     */
+    private String getContainerInfo(String serviceName, String projectName) {
+        List<String> dockerCommands = new ArrayList<>();
+        dockerCommands.add("docker");
+        dockerCommands.add("ps");
+
+        // Filter containers by docker-compose project name label
+        if (StringUtils.isNotEmpty(projectName)) {
+            dockerCommands.add("-f");
+            dockerCommands.add("label=com.docker.compose.project=" + projectName);
+        }
+
+        // Filter containers by docker-compose service name label
+        dockerCommands.add("-f");
+        dockerCommands.add("label=com.docker.compose.service=" + serviceName);
+
+        // Format output to only includes name and ports section of container
+        dockerCommands.add("--format");
+        dockerCommands.add("{{.Names}}#{{.Ports}}");
+
+        // Execute command and make the result
+        ProcessExecutor executor = new ProcessExecutor()
+                .environment(environment);
+        List<String> results = execute(executor, dockerCommands).getOutput().getLines();
+        StringBuilder resultBuilder = new StringBuilder();
+        if (results != null && !results.isEmpty()) {
```

Comment:
There should be just one line in output.
Assert otherwise (if there is no line or multiple one).
Also throw AssertionError if in that line there no "#" or more than one instances of "#"

Reply:
As we talked ;)